core-bundle/src/Resources/contao/controllers/FrontendShare.php

```diff
@@ -26,20 +26,28 @@ class FrontendShare extends Frontend
 	 */
 	public function run()
 	{
-		switch (Input::get('p'))
+		$url = Input::get('u', true);
+
+		if (!$url || !\is_string($url))
+		{
+			return new RedirectResponse('../');
+		}
+
+		if (Input::get('p') == 'facebook')
+		{
+			return new RedirectResponse('https://www.facebook.com/sharer/sharer.php?p[url]=' . rawurlencode($url));
+		}
+
+		$text = Input::get('t', true);
+
+		if (!$text || !\is_string($text))
+		{
+			return new RedirectResponse('../');
+		}
```

Comment:
is a text really mandatory to share this on twitter? Can't I just share the URL and keep text empty?

Reply:
Changed in 3e909d1645305bb73907c55e75b9bc13f377796a.